### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by zemola**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

### Create custom methods
 models include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method that returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

Additionally, you will need to make current user data available to all controllers.

-  Create a method called current_user in your ApplicationController.
-  A current_user method returns the first user from a database.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_counter
+    user.increment!(:posts_counter)
```
Comment (hiromataba) [inline app/models/post.rb]: - [ ] You should use `author.increment!(:posts_counter)` not user

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_posts_counter
```
Comment (hiromataba) [inline app/models/post.rb:10]: - [ ] [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private

`after_save :update_post_counter`

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @zemola    :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Very professional README file :+1: 
- Nice commit history :heavy_check_mark: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
Hi **@[zemola](https://github.com/zemola),** 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
appr

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**